src/cargo/util/hostname.rs

```diff
@@ -0,0 +1,85 @@
+use std::{ffi::OsString, io};
```

Comment:
Copied from [ripgrep](https://github.com/BurntSushi/ripgrep/blob/7099e174acbcbd940f57e4ab4913fee4040c826e/crates/cli/src/hostname.rs) because the `hostname` and `gethostname` packages had what looked like less robust implementations.

Reply:
Done